mu-plugins/10up-plugin/src/Assets.php

```diff
@@ -0,0 +1,273 @@
+	}
+
+	/**
+	 * Enqueue scripts for front-end.
+	 *
+	 * @return void
+	 */
+	public function scripts() {
+
+		wp_enqueue_script(
+			'tenup_plugin_shared',
+			$this->script_url( 'shared', 'shared' ),
+			$this->get_asset_info( 'shared', 'dependencies' ),
+			$this->get_asset_info( 'shared', 'version' ),
+			true
+		);
+
+		wp_enqueue_script(
+			'tenup_plugin_frontend',
+			$this->script_url( 'frontend', 'frontend' ),
+			$this->get_asset_info( 'frontend', 'dependencies' ),
+			$this->get_asset_info( 'frontend', 'version' ),
+			true
+		);
+	}
+
+	/**
+	 * Enqueue scripts for admin.
+	 *
+	 * @return void
+	 */
+	public function admin_scripts() {
+
+		wp_enqueue_script(
+			'tenup_plugin_shared',
+			$this->script_url( 'shared', 'shared' ),
+			$this->get_asset_info( 'shared', 'dependencies' ),
+			$this->get_asset_info( 'shared', 'version' ),
+			true
+		);
+
+		wp_enqueue_script(
+			'tenup_plugin_admin',
+			$this->script_url( 'admin', 'admin' ),
+			$this->get_asset_info( 'admin', 'dependencies' ),
+			$this->get_asset_info( 'admin', 'version' ),
+			true
+		);
+	}
+
+	/**
+	 * Enqueue styles for front-end.
+	 *
+	 * @return void
+	 */
+	public function styles() {
+
+		wp_enqueue_style(
+			'tenup_plugin_shared',
+			$this->style_url( 'shared', 'shared' ),
+			[],
+			$this->get_asset_info( 'shared', 'version' ),
+		);
+
+		if ( is_admin() ) {
+			wp_enqueue_style(
+				'tenup_plugin_admin',
+				$this->style_url( 'admin', 'admin' ),
+				[],
+				$this->get_asset_info( 'admin', 'version' ),
+			);
+		} else {
+			wp_enqueue_style(
+				'tenup_plugin_frontend',
+				$this->style_url( 'frontend', 'frontend' ),
+				[],
+				$this->get_asset_info( 'frontend', 'version' ),
+			);
+		}
+	}
+
+	/**
+	 * Enqueue styles for admin.
+	 *
+	 * @return void
+	 */
+	public function admin_styles() {
+
+		wp_enqueue_style(
+			'tenup_plugin_shared',
+			$this->style_url( 'shared', 'shared' ),
+			[],
+			$this->get_asset_info( 'shared', 'version' ),
+		);
+
+		wp_enqueue_style(
+			'tenup_plugin_admin',
+			$this->style_url( 'admin', 'admin' ),
+			[],
+			$this->get_asset_info( 'admin', 'version' ),
+		);
+	}
+
+	/**
+	 * Enqueue editor styles. Filters the comma-delimited list of stylesheets to load in TinyMCE.
+	 *
+	 * @param string $stylesheets Comma-delimited list of stylesheets.
+	 * @return string
+	 */
+	public function mce_css( $stylesheets ) {
+		if ( ! empty( $stylesheets ) ) {
+			$stylesheets .= ',';
+		}
+
+		return $stylesheets . TENUP_PLUGIN_URL . ( ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ?
+				'assets/css/frontend/editor-style.css' :
+				'dist/css/editor-style.min.css' );
+	}
+
+	/**
+	 * Add async/defer attributes to enqueued scripts that have the specified script_execution flag.
+	 *
+	 * @link https://core.trac.wordpress.org/ticket/12009
+	 * @param string $tag    The script tag.
+	 * @param string $handle The script handle.
+	 * @return string|null
+	 */
+	public function script_loader_tag( $tag, $handle ) {
```

Comment:
Wasn't this supported a while ago? https://make.wordpress.org/core/2023/07/14/registering-scripts-with-async-and-defer-attributes-in-wordpress-6-3/

Reply:
Agreed, I don't think this is the right place to do that though. This is primarily around changing the structure of the scaffold, so I'm trying to move things 1:1. Let's spin up an issue to do this though!